### ISSUE TITLE
Upgrading from 2.18 to 3.2 is not triggering the update dialog. 

### ISSUE THREAD

**Issue by newtewt** [medic/medic-webapp#5247]
When going from 2.18 to 3.x the dialog to update isn't presented to a user. The user is unaware the app updated and would need to do a manual refresh to get the latest app changes.  I'm not sure how keen we are to releasing a 2.18.2 as there is a simple work around to this issue.  @garethbowen  @abbyad 

**Steps to reproduce**:
Have an instance with 2.18.
Log in as an offline user in chrome.
Ensure replication completes.
Check the offline box in dev tools network tab.
Upgrade the instance to 3.2.x.
Once upgrade completes uncheck the offline checkbox.
Wait for the replication to occur.
The requests for the new app-cache, settings, and medic-client all occur in the network tab.

**What should happen**:
The dialog to update should be displayed and the user gets the updated app.

**What actually happens**:
The modal to update is never shown.
Manually refreshing the browser pulls the updated app.

See here for @dianabarsan  results from her investigation. 

https://github.com/medic/medic-webapp/issues/5235#issuecomment-455198314

**Environment**:

- Instance: local environment using horti
- Browser: Chrome
- Client platform: Ubuntu
- App: webapp
- Version: 2.18 -> 3.2.X

**Comment by garethbowen** [comment]
I'm not keen to do a 2.18.2 release just for this but I think we can fix it in 3.2.x, possibly with a redirect from the old manifest to the new one in api. I'll do some investigation on this and try to get it in to 3.2.1.

**Comment by garethbowen** [comment]
We have a [Changes listener in the 2.18 code](https://github.com/medic/medic-webapp/blob/2.18.x/static/js/controllers/inbox.js#L634-L654) which checks for modified docs with any of these IDs: '_design/medic', '_design/medic-client', or 'appcache'. When a change is noticed in any of these the user should be prompted to refresh. *All* of these files should be modified on upgrade.

I've tested in development that changing these files prompts the dialog for me, as expected.

@newtewt Are you sure replication had completed? Can you include the console output? I expect some helpful errors to be shown.

**Comment by dianabarsan** [comment]
@garethbowen 
I also replicated the issue locally. You can trigger it by using a user that has no docs (just it's contact and place).

**Comment by garethbowen** [comment]
After re-reading Diana's analysis I think I see the problem - it needs multiple refreshes. One marks the appcache as obsolete, the next one downloads the new content, and finally one to load the new content.

**Comment by dianabarsan** [comment]
Yea, appcache needs 2 updates (one obsolete + one fail), the browser window only needs one refresh.

**Comment by garethbowen** [comment]
We can't move the manifest back because that would break for anyone on 3.0-3.2. 

We only need to get through a couple of minor revisions until this is made obsolete by service workers (  #4713 ) but it's not stable enough to consider backporting yet.

I'm wondering if we can just serve the manifest.appcache from both locations. Because the contents have relative URLs we'd need to have two different versions of the manifest but that's not too difficult. Then they both go away when service workers are available.

**Comment by garethbowen** [comment]
Ok, I think that solution works (PR incoming!). Now what happens is...

1. The request for the existing appcache finds a change instead of firing an "obsolete" event
2. This change downloads all the _new_ files into their existing appcache
3. They are then prompted to refresh and after they do everything works correctly with the new code
4. The new code includes the HTML referencing the new appcache location and new JS which detects when the appcache is UNCACHED. On this load the new cache is UNCACHED so the user is prompted to refresh to populate it.
5. Next time they refresh the new appcache downloads the new code, and the user is prompted to refresh a third time.
6. Everything works as expected.

The downside of this approach is obviously that the user is prompted three times, but that's better than not being prompted at all. This all goes away once we get service workers merged...

AT on this is complicated but make sure that once the prompts have stopped coming you can go offline and reload the app as expected.

**Comment by newtewt** [comment]
This is working as expected now. In browser and on phone. The offline user gets prompted to update after the app has been updated. They get the latest changes as well. 

However, in firefox, if you're logged in as admin and watching the update. The Update dialog is displayed after the app updates. Click update refreshes the page and keeps displaying the update dialog. After pressing update at least 10 times it seems to not go away. Cancelling the dialog the user is returned to the 404 error page because the admin console has moved.  This is not an issue if you're on the admin page and refresh in chrome. 

I tried going from 2.18 to 3.0 and 2.18 to 3.2.0 and they worked fine in Firefox.  This seems to have been introduced with changes in 3.2.1.beta.2. 

@abbyad  @garethbowen  do we want to hold this open or move forward?
